### PR TITLE
refactor(aws): Refactor provider

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Refactor code from AWS provider to ensure that it can be used as a library from the SDK.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/aws_provider.py [hunk 1]
```diff
@@ -64,27 +64,50 @@ class AwsProvider(Provider):
...
+    def __init__(
+        self,
+        aws_retries_max_attempts=3,
+        input_role=None,
+        input_session_duration=None,
+        input_external_id=None,
+        input_role_session_name=None,
+        input_mfa=None,
+        input_profile=None,
+        input_regions=None,
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py]:
```suggestion
        input_regions=set(),
```

File: prowler/providers/aws/aws_provider.py [hunk 2]
```diff
@@ -64,27 +64,50 @@ class AwsProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments: Namespace):
+    def __init__(
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py:67]: I think it is a good time to rename the arguments to something better like:
- `input_role` -> `role_arn`
- `input_external_id` -> `external_id`

File: tests/providers/aws/aws_provider_test.py [hunk 3]
```diff
@@ -249,7 +250,10 @@ def test_aws_provider_default(self):
         arguments = Namespace()
         arguments.mfa = False
         arguments.scan_unused_services = True
```
Comment (jfagoagas) [inline tests/providers/aws/aws_provider_test.py]: You can remove all this references in this file because namespace is no longer needed.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/aws_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py]: Done!

File: prowler/providers/aws/aws_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py:67]: Done!

File: tests/providers/aws/aws_provider_test.py (hunk 3 above)
Comment (pedrooot) [reply tests/providers/aws/aws_provider_test.py]: Done!

File: prowler/providers/common/provider.py [hunk 4]
```diff
@@ -180,7 +180,25 @@ def set_global_provider(arguments):
             )
 
             if not isinstance(Provider._global, provider_class):
-                global_provider = provider_class(arguments)
+                if "aws" in provider_class_name.lower():
+                    global_provider = provider_class(
+                        arguments.retries_max_attempts,
```
Comment (jfagoagas) [inline prowler/providers/common/provider.py]:
```suggestion
                        arguments.aws_retries_max_attempts,
```

File: tests/lib/cli/parser_test.py [hunk 5]
```diff
@@ -1064,12 +1064,12 @@ def test_aws_parser_wrong_resource_arn(self):
...
         assert ex.type == SystemExit
 
-    def test_aws_parser_aws_retries_max_attempts(self):
+    def test_aws_parser_retries_max_attempts(self):
         argument = "--aws-retries-max-attempts"
         max_retries = "10"
         command = [prowler_command, argument, max_retries]
         parsed = self.parser.parse(command)
-        assert parsed.aws_retries_max_attempts == int(max_retries)
+        assert parsed.retries_max_attempts == int(max_retries)
```
Comment (jfagoagas) [inline tests/lib/cli/parser_test.py]:
```suggestion
        assert parsed.aws_retries_max_attempts == int(max_retries)
```

**Review by jfagoagas — DISMISSED** [review]
🚀

**Review by jfagoagas — DISMISSED** [review]

**Review by jfagoagas — DISMISSED** [review]

File: prowler/providers/common/provider.py [hunk 6]
```diff
@@ -180,6 +180,23 @@ def set_global_provider(arguments):
...
+                        arguments.profile,
+                        set(arguments.region) if arguments.region else None,
+                        arguments.organizations_role,
+                        arguments.scan_unused_services,
+                        arguments.resource_tag,
+                        arguments.resource_arn,
+                        arguments.config_file,
+                        arguments.fixer_config,
+                    )
                 if "Kubernetes" in provider_class_name:
```
Comment (jfagoagas) [inline prowler/providers/common/provider.py]:
```suggestion
                elif "Kubernetes" in provider_class_name:
```

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**